var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -97,21 +97,24 @@ def _build_mods(self, mods_location, link_flag='', include_flag='',
                                mods_location)
 
         assert os.path.isfile(os.path.join(output_dir, 'special'))
-        return include_flag, link_flag
+        return include_flag_raw, link_flag_raw
+
+    def _nrnivmodlcore_params(self, include_flag, link_flag):
+        return ['-n', self.mech_name, '-i', include_flag, '-l', link_flag, '-V']
+
+    def _add_corenrn_include_flag(self):
+        return ' -DENABLE_CORENEURON' \
+            + ' -I%s' % self.spec['coreneuron'].prefix.include
 
     def __build_mods_coreneuron(self, mods_location, link_flag, include_flag):
         mods_location = os.path.abspath(mods_location)
         assert os.path.isdir(mods_location) and find(mods_location, '*.mod',
                                                      recursive=False),\
             'Invalid mods dir: ' + mods_location
-        nrnivmodl_params = ['-n', self.mech_name,
-                            '-i', include_flag,
-                            '-l', link_flag,
-                            '-V',
-                            'mod']
+        nrnivmodl_params = self._nrnivmodlcore_params(include_flag, link_flag)
         with working_dir('build_' + self.mech_name, create=True):
             force_symlink(mods_location, 'mod')
-            which('nrnivmodl-core')(*nrnivmodl_params)
+            which('nrnivmodl-core')(*(nrnivmodl_params + ['mod']))
```

Comment:
@ferdonline : I am not sure where we are removing non-coreneuron compatible mod files. We were saying we will do:
```
cp -r mod mod_core
rm mod_core/*incompatible-mods-*.mod
nrnivmodl-core mod_core
nrnivmodl mod "-L x86/64 -lcorenrnmech"

Reply:
Yep, logic not there yet :)

Reply:
Something additional to that, I am wondering what should happen if someone wants to edit some of the existing modfiles and compile it with `build_neurodamus.sh`. If I am not mistaken, with the current way of linking `libnrnmech.so` in the script this is not possible without changing the name of the mod file. Is this something we should somehow handle?

Reply:
we can sync on this tomorrow after standup.

Reply:
Also not totally sure I understand. We can discuss tomorrow :)

Reply:
where `neuron_only_mods.txt` comes from? i.e. how we know which mod files to be excluded for nrnivodl-core?